pennylane/pauli/conversion.py

```diff
@@ -400,17 +411,26 @@ def _(op: Hamiltonian):
     if not all(is_pauli_word(o) for o in op.ops):
         raise ValueError(f"Op must be a linear combination of Pauli operators only, got: {op}")
 
-    summands = []
+    def term_2_pauli_word(term):
+        if isinstance(term, Tensor):
+            pw = dict((obs.wires[0], obs.name[-1]) for obs in term.non_identity_obs)
+        elif isinstance(term, Identity):
+            pw = {}
+        else:
+            pw = dict([(term.wires[0], term.name[-1])])
+        return PauliWord(pw)
+
+    ps = PauliSentence()
     for coeff, term in zip(*op.terms()):
-        ps = _pauli_sentence(term)
-        for pw, sub_coeff in ps.items():
-            ps[pw] = coeff * sub_coeff
-        summands.append(ps)
+        sub_ps = PauliSentence({term_2_pauli_word(term): coeff})
+        ps += sub_ps
 
-    return reduce(lambda a, b: a + b, summands) if len(summands) > 0 else PauliSentence()
+    return ps
 
 
 @_pauli_sentence.register
 def _(op: Sum):
-    summands = (_pauli_sentence(summand) for summand in op)
-    return reduce(lambda a, b: a + b, summands)
+    ps = PauliSentence()
+    for term in op:
+        ps += _pauli_sentence(term)
+    return ps
```

Comment:
i'm actually not sure why this exists given that the pauli-rep should have been pre-computed. or at least, it should just be `return op._pauli_rep`

Reply:
Can't you have a sum of operators that don't have a Pauli rep? This is just going to crash later on, but then we'll know which is the offending term, no?

Reply:
Seems like `return op._pauli_rep` will return `None`, but unit tests indicate the expected behaviour is to throw a `ValueError`.

Reply:
I'll call it out-of-scope for this PR then, your changes look good 👍